indico/modules/users/util.py

```diff
@@ -557,3 +559,26 @@ def send_avatar(user):
 def get_avatar_url_from_name(first_name):
     first_char = first_name[0] if first_name else None
     return url_for('assets.avatar', name=first_char)
+
+
+@memoize_request
+def get_mastodon_server_info(url):
+    """Gets the server name and domain of a Mastodon instance."""
+    server_url = strip_path_from_url(url)
+
+    if not server_url:
+        return None
+
+    try:
+        resp = requests.get(f'{server_url}/api/v2/instance')
+        resp.raise_for_status()
+    except requests.RequestException:
+        return None
+    try:
+        data = resp.json()
```

Comment:
I think this may raise an error that's not handled if you get a successful response that's not valid JSON

Reply:
And indeed it did. Already fixed though.